src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128_1.cs

```diff
@@ -116,6 +116,15 @@ public bool Equals(Vector128<T> other)
                 return Sse.MoveMask(result) == 0b1111; // We have one bit per element
             }
 
+            if (Sse41.IsSupported && (typeof(T) != typeof(double)))
```

Comment:
nit: I think it would be better to do something like the below, as it results in a bit simpler (IMO) codeflow:
```
if (Sse2.IsSupported)
{
    if (typeof(T) == typeof(double))
    {
    }
    else if (Sse41.IsSupporteD)
    {
    }
    else
    {
    }
}
```